services/horizon/internal/actions/orderbook.go

```diff
@@ -0,0 +1,147 @@
+package actions
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/stellar/go/amount"
+	"github.com/stellar/go/exp/orderbook"
+	protocol "github.com/stellar/go/protocols/horizon"
+	"github.com/stellar/go/services/horizon/internal/resourceadapter"
+	"github.com/stellar/go/support/render/problem"
+	"github.com/stellar/go/xdr"
+)
+
+// StreamableObjectResponse is an interface for objects returned by streamable object endpoints
+// A streamable object endpoint is an SSE endpoint which returns a single JSON object response
+// instead of a page of items.
+type StreamableObjectResponse interface {
+	Equals(other StreamableObjectResponse) bool
+}
+
+// OrderBookResponse is the response for the /orderbook_endpoint
+// OrderBookResponse implements StreamableObjectResponse
+type OrderBookResponse struct {
+	protocol.OrderBookSummary
+}
+
+func priceLevelsEqual(a, b []protocol.PriceLevel) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	for i := range a {
+		if a[i] != b[i] {
+			return false
+		}
+	}
+
+	return true
+}
+
+// Equals returns true if the OrderBookResponse is equal to `other`
+func (o OrderBookResponse) Equals(other StreamableObjectResponse) bool {
+	otherOrderBook, ok := other.(OrderBookResponse)
+	if !ok {
+		return false
+	}
+	return otherOrderBook.Selling == o.Selling &&
+		otherOrderBook.Buying == o.Buying &&
+		priceLevelsEqual(otherOrderBook.Bids, o.Bids) &&
+		priceLevelsEqual(otherOrderBook.Asks, o.Asks)
+}
+
+var invalidOrderBook = problem.P{
+	Type:   "invalid_order_book",
+	Title:  "Invalid Order Book Parameters",
+	Status: http.StatusBadRequest,
+	Detail: "The parameters that specify what order book to view are invalid in some way. " +
+		"Please ensure that your type parameters (selling_asset_type and buying_asset_type) are one the " +
+		"following valid values: native, credit_alphanum4, credit_alphanum12.  Also ensure that you " +
+		"have specified selling_asset_code and selling_asset_issuer if selling_asset_type is not 'native', as well " +
+		"as buying_asset_code and buying_asset_issuer if buying_asset_type is not 'native'",
+}
+
+// GetOrderbookHandler is the action handler for the /order_book endpoint
+type GetOrderbookHandler struct {
+	OrderBookGraph *orderbook.OrderBookGraph
+}
+
+func offersToPriceLevels(offers []xdr.OfferEntry, invert bool) []protocol.PriceLevel {
+	result := []protocol.PriceLevel{}
+
+	amountForPrice := map[xdr.Price]xdr.Int64{}
+	for _, offer := range offers {
+		amountForPrice[offer.Price] += offer.Amount
+	}
+	for _, offer := range offers {
+		total, ok := amountForPrice[offer.Price]
+		if !ok {
+			continue
+		}
+
+		offerPrice := offer.Price
+		if invert {
+			offerPrice.Invert()
+		}
+
+		result = append(result, protocol.PriceLevel{
+			PriceR: protocol.Price{
+				N: int32(offerPrice.N),
+				D: int32(offerPrice.D),
+			},
+			Price:  offerPrice.String(),
+			Amount: amount.String(total),
+		})
+
+		delete(amountForPrice, offerPrice)
```

Comment:
Shouldn't we use `offer.Price` here? `offerPrice` can be inverted.
```suggestion
		delete(amountForPrice, offer.Price)
```